symengine/sets.cpp

```diff
@@ -766,7 +766,10 @@ bool ImageSet::__eq__(const Basic &o) const
 {
     if (is_a<ImageSet>(o)) {
         const ImageSet &other = down_cast<const ImageSet &>(o);
-        return unified_eq(sym_, other.sym_) and unified_eq(expr_, other.expr_)
+        map_basic_basic d;
+        d[sym_]
+            = other.sym_; // ImageSet(x,x**2,base) == ImageSet(y, y**2, base)
+        return unified_eq(expr_->subs(d), other.expr_)
```

Comment:
```
soln->__str__() = {2*pi*_n + pi | _n in (-oo, oo)} U {2*pi*_n | _n in (-oo, oo)}
req->__str__() = {2*pi*_n + pi | _n in (-oo, oo)} U {2*pi*_n | _n in (-oo, oo)}
```
both are not equal as dummy in testing module and in solve module are different.
One possible workaround could be replacing dummy `_n` with symbol `n` in `solve`  and testing with the same in `test_solve.cpp`